packages/experimental/TabList/src/TabList/useTabList.ts

```diff
@@ -53,9 +54,17 @@ export const useTabList = (props: TabListProps): TabListInfo => {
     [setTabKeys],
   );
 
+  // Logic to style the animated indicator
+  const { addTabLayout, onTabListLayout, tabListLayout, styles, updateStyles } = useAnimatedIndicator(
```

Comment:
Why is the tab list generating the animated indicator's styles? shouldn't the indicator do that itself?

Reply:
Initially, I tried calling the hook inside the indicator. The hook needs to know the selected key and whether the TabList is vertical or horizontal. This is info passed through the TabList context, but the issue is that when trying to access the context in the indicator component, the context never updates - its values are the same as they are when we initially define it via `React.createContext()`. The indicator is also a child of the context provider, unless I'm hallucinating which makes this issue all the more strange. 

The hook also has to run in the first render, so my work around is to run it in the TabList component and pass its results to the indicator.

Reply:
That sounds like a bug that you should look into. To me that would mean that the context provider isn't getting the value prop as a new object when the properties update.

Reply:
It's strange because the TabList context inside of the Tab component updates as expected. It's just for this specific component. I'm not really sure what to make of this bug.

Reply:
@rurikoaraki Did some investigation - it seems like the bug comes from calling `useContext` inside a slot component which uses the framework (`stagedComponent`, `compose`, `compressible`). Making the indicator a non-slot component fixes the issue, so I was able to move the styling logic into the `useAnimatedIndicator` hook